### PR TITLE
Support for bundle minification & production build

### PR DESCRIPTION
**Pull request by Gebov**

### CODE REVIEW

File: config/webpack.config.js [hunk 1]
```diff
@@ -4,46 +4,57 @@ const constants = require("./constants");
...
-module.exports = {
-    cache: true,
-    devtool: "source-map",
-    entry: {
-        [extensionsKey]: extensionsFileName
-    },
+module.exports = function(env, argv) {
+    const config = {
+        cache: true,
+        devtool: "source-map",
```
Comment (vassildinev) [inline config/webpack.config.js]: Should we have source map in prod?

File: config/import.plugin.js [hunk 2]
```diff
@@ -58,7 +58,7 @@ ImportPlugin.prototype.apply = function (compiler) {
             const indexOfExports = originalSource.indexOf("exports.", indexOfMethod);
             const indexOfDelimeter = originalSource.indexOf(";", indexOfExports);
 
-            const codeToInject = `exports.metadata = { version: ${JSON.stringify(require("../package.json").devDependencies["progress-sitefinity-adminapp-sdk"])}, name: "${constants.extensionsKey}" };`;
+            const codeToInject = `arguments[1].metadata = { version: ${JSON.stringify(require("../package.json").devDependencies["progress-sitefinity-adminapp-sdk"])}, name: "${constants.extensionsKey}" };`;
```
Comment (vassildinev) [inline config/import.plugin.js]: Can you please explain?

Comment (Gebov) [reply config/import.plugin.js]: Yes, when the source code is minified, the exports keyword is not available. To ensure that this works both with minified and unminified code I read the second argument

File: config/webpack.config.js (hunk 1 above)
Comment (Gebov) [reply config/webpack.config.js]: fixed

**Pull request closed without merging**